deploy/cephfs/helm/templates/provisioner-deployment.yaml

```diff
@@ -30,7 +30,8 @@ spec:
           image: "{{ .Values.provisioner.image.repository }}:{{ .Values.provisioner.image.tag }}"
           args:
             - "--csi-address=$(ADDRESS)"
-            - "--leader-election-type=leases"
+            - "--enable-leader-election=true"
+            - "--leader-election-type=endpoints"
```

Comment:
Why are we moving away from leases? Kubernetes CSI seems to be suggesting that this option itself would be [deprecated](https://github.com/kubernetes-csi/external-provisioner/blob/master/README.md#deprecated-arguments) in the future as the election mechanism would default to leases, so why are we starting with `endpoints` and what are the repercussions.

Reply:
@ShyamsundarR `external-provisoner` is failing with `leases` https://github.com/kubernetes-csi/external-provisioner/issues/291

Reply:
`endpoints` seem to have a request flooding problem, as noted [here](https://github.com/kubernetes/kubernetes/issues/68819#issuecomment-423098702). In consideration of this should we even proceed with `endpoints` or move back to Stateful sets, or should we wait for a fix/ack for the Kubernetes issue?

Reply:
seems like `leases` works with `kube 1.14+` version.  do we need to switch back to statefulset to support older kubernetes version or will use endpoint as lease?

one question needs to answer here is how many systems will be affected with throttling enabled due to the endpoint?

Reply:
CI passed with `lease` and kube 1.14 https://travis-ci.org/ceph/ceph-csi/jobs/545169705

Reply:
> seems like `leases` works with `kube 1.14+` version. do we need to switch back to statefulset to support older kubernetes version or will use endpoint as lease?

I think it is too early to drop v1.13 support, so either have 2 deployments (one for each version), or switch back to Stateful sets (not optimal for future requirements of upgrade etc.).

> 
> one question needs to answer here is how much systems will be affected with throttling enabled due to the endpoint?

This [issue](https://github.com/kubernetes-csi/external-provisioner/issues/68) and its related discussion around 100PVCs seems to suggest this could happen at lower threasholds, I would hence be wary of using the `endpoint` leadership election mechanism till we know better.

I would say have different deployments for different Kubernetes versions, and hence use leases only for v1.14 and Stateful sets for v1.13 is a solution that we should go for.